libbeat/autodiscover/autodiscover.go

```diff
@@ -207,19 +206,27 @@ func (a *Autodiscover) handleStart(event bus.Event) bool {
 
 		// Update meta no matter what
 		dynFields := a.meta.Store(hash, meta)
-
-		if a.configs[eventID][hash] != nil {
+		// If a config is already present and hasn't changed then add it to new config so that it continues to run.
+		// If not, then create a new config and add it in to the new list.
+		if cfg, ok := oldCfg[hash]; ok {
 			a.logger.Debugf("Config %v is already running", config)
+			newCfg[hash] = cfg
 			continue
-		}
-
-		a.configs[eventID][hash] = &reload.ConfigWithMeta{
-			Config: config,
-			Meta:   &dynFields,
+		} else {
+			newCfg[hash] = &reload.ConfigWithMeta{
+				Config: config,
+				Meta:   &dynFields,
+			}
 		}
 		updated = true
 	}
 
+	// By replacing the config's for eventID we make sure that all old configs that are no longer in use
+	// are stopped correctly. This will ensure that a resync event is handled correctly.
+	if updated {
+		a.configs[eventID] = newCfg
+	}
```

Comment:
Something I just realized about this is that it may affect events from the same container when exposing multiple ports. In these cases we spawn several events (1 per exposed port), which could lead to some configs being lost in the wild, right?

Reply:
i have fixed this by making sure that all ports' worth events (which all have the same id) are combined to one event.